src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -754,41 +754,46 @@ public SendCurrentClusterState(IActorRef receiver)
         }
 
         /// <summary>
-        /// TBD
+        /// INTERNAL API.
+        ///
+        /// Marker interface for publication events from Akka.Cluster.
         /// </summary>
-        interface IPublishMessage { }
+        /// <remarks>
+        /// <see cref="INoSerializationVerificationNeeded"/> is not explicitly used on the JVM,
+        /// but without it we run into serialization issues via https://github.com/akkadotnet/akka.net/issues/3724
+        /// </remarks>
+        private interface IPublishMessage : INoSerializationVerificationNeeded { }
```

Comment:
First fix: made it so `IPublishMessage` doesn't get checked by `akka.actor.serialize-messages` - did this because designating a single `Gossip` class constructor with `JsonConstructor` wasn't going to work due to the fact that we make use of all three different constructors in different situations. The JVM gets away with this by using the built-in `Serializable` trait in Scala, which is roughly equivalent to our binary formatter, which .NET no longer has in .NET Core. So, I figured this would be the best substitute here.

Reply:
Relevant link on Scala behavior: https://alvinalexander.com/scala/how-to-use-serialization-in-scala-serializable-trait